File: src/main/java/com/braggbnbsevenseventyseven/service/GuideService.java
package com.braggbnbsevenseventyseven.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnbsevenseventyseven.domain.Guide;
import com.braggbnbsevenseventyseven.dto.GuideDTO;
import com.braggbnbsevenseventyseven.dto.GuideSearchDTO;
import com.braggbnbsevenseventyseven.dto.GuidePageDTO;
import com.braggbnbsevenseventyseven.dto.GuideConvertCriteriaDTO;
import com.braggbnbsevenseventyseven.service.GenericService;
import com.braggbnbsevenseventyseven.dto.common.RequestDTO;
import com.braggbnbsevenseventyseven.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface GuideService extends GenericService<Guide, Integer> {

	List<Guide> findAll();

	ResultDTO addGuide(GuideDTO guideDTO, RequestDTO requestDTO);

	ResultDTO updateGuide(GuideDTO guideDTO, RequestDTO requestDTO);

    Page<Guide> getAllGuides(Pageable pageable);

    Page<Guide> getAllGuides(Specification<Guide> spec, Pageable pageable);

	ResponseEntity<GuidePageDTO> getGuides(GuideSearchDTO guideSearchDTO);
	
	List<GuideDTO> convertGuidesToGuideDTOs(List<Guide> guides, GuideConvertCriteriaDTO convertCriteria);

	GuideDTO getGuideDTOById(Integer guideId);







}

File: braggbnbsevenseventyseven_ui/src/services/GuideService.js
import http from "../http-common"; 

class GuideService {
  getAllGuides(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/guide/guides`, searchDTO);
  }

  get(guideId) {
    return this.getRequest(`/guide/${guideId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/guide?field=${matchData}`, null);
  }

  addGuide(data) {
    return http.post("/guide/addGuide", data);
  }

  update(data) {
  	return http.post("/guide/updateGuide", data);
  }
  
  uploadImage(data,guideId) {
  	return http.postForm("/guide/uploadImage/"+guideId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new GuideService();
